File: Examples/test-suite/python/exception_order_runme.py
from exception_order import *

# This test is expected to fail with -builtin option.
# Throwing builtin classes as exceptions not supported
if is_python_builtin():
    exit(0)

a = A()

try:
    a.foo()
except E1 as e:
    pass
except:
    raise RuntimeError("bad exception order")

try:
    a.bar()
except E2 as e:
    pass
except:
    raise RuntimeError("bad exception order")

try:
    a.foobar()
except RuntimeError as e:
    if e.args[0] != "postcatch unknown":
        raise RuntimeError("bad exception order {}".format(e.args))


try:
    a.barfoo(1)
except E1 as e:
    pass
except:
    raise RuntimeError("bad exception order")

try:
    a.barfoo(2)
except E2 as e:
    pass
except:
    raise RuntimeError("bad exception order")

File: Examples/test-suite/matlab/exception_order_runme.m
import exception_order.*

a = A();

try
  a.foo()
catch
  exception_order_helper(lasterror, 'C++ side threw an exception of type E1');
end

try
  a.bar()
catch
  exception_order_helper(lasterror, 'C++ side threw an exception of type E2');
end

try
  a.foobar()
catch
  exception_order_helper(lasterror, 'postcatch unknown (SWIG_RuntimeError)');
end

try
  a.barfoo(1)
catch
  exception_order_helper(lasterror, 'C++ side threw an exception of type E1');
end

try
  a.barfoo(2)
catch
  exception_order_helper(lasterror, 'C++ side threw an exception of type E2 *');
end
